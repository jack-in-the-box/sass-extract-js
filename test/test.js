const path = require('path');
const sassExtract = require('sass-extract');
const createPlugin = require('../lib/plugin');

const testfiles = {
  basic: path.resolve(__dirname, './sass/test-basic.scss'),
  camel: path.resolve(__dirname, './sass/test-opts-camel.scss'),
  sassOpts: path.resolve(__dirname, './sass/test-opts-sass.scss'),
  import: path.resolve(__dirname, './sass/test-import.scss'),
  empty: path.resolve(__dirname, './sass/test-empty.scss'),
};

const getVars = (file, compileOpts, plugin) => sassExtract.renderSync(
  { file, ...compileOpts || {}},
  { plugins: [plugin || path.resolve('../sass-extract-js')] },
).vars;

describe('sass-extract-js', () => {
  it('should convert basic SASS vars', () => {
    expect(getVars(testfiles.basic)).toMatchSnapshot();
  });

  it('should convert keys to camelCase', () => {
    expect(getVars(testfiles.camel)).toMatchSnapshot();
  });

  it('should handle imports', () => {
    expect(getVars(testfiles.import)).toMatchSnapshot();
  });

  it('should handle file with no variables', () => {
    expect(getVars(testfiles.empty)).toMatchSnapshot();
  });

  it('should pass sass options through', () => {
    const opts = {
      includePaths: [path.join(__dirname, 'sass', 'nested')],
    };
    expect(getVars(testfiles.sassOpts, opts)).toMatchSnapshot();
  });

  describe('options', () => {
    const shouldNotConvert = (pluginInst) => (
      it('should NOT convert keys to camelCase', () => {
        expect(getVars(testfiles.camel, null, pluginInst)).toMatchSnapshot();
      })
    );

    describe('plugin instance with options', () => {
      const inst = createPlugin({ camelCase: false });
      shouldNotConvert(inst);
    });

    describe('plugin instance without options', () => {
      const inst = {
        plugin: createPlugin(),
        options: { camelCase: false },
      };
      shouldNotConvert(inst);
    });

    describe('plugin name', () => {
      const inst = {
        plugin: path.resolve('../sass-extract-js'),
        options: { camelCase: false },
      };
      shouldNotConvert(inst);
    });
  });
});
