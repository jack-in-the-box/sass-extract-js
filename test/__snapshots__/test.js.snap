// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sass-extract-js options plugin instance with options should NOT convert keys to camelCase 1`] = `
{
  "MY-VAR": "500px",
  "My-Cool_var": "400px",
  "my-var": "200px",
  "myVar": 0,
  "my__var--3": "700px",
  "my_other_vAr": "300px",
  "my_var": "200px",
  "my_var_2": "600px",
  "value-Map": {
    "my-var5": "900px",
    "my__var--6": "1000px",
    "my_var4": "800px",
  },
}
`;

exports[`sass-extract-js options plugin instance without options should NOT convert keys to camelCase 1`] = `
{
  "MY-VAR": "500px",
  "My-Cool_var": "400px",
  "my-var": "200px",
  "myVar": 0,
  "my__var--3": "700px",
  "my_other_vAr": "300px",
  "my_var": "200px",
  "my_var_2": "600px",
  "value-Map": {
    "my-var5": "900px",
    "my__var--6": "1000px",
    "my_var4": "800px",
  },
}
`;

exports[`sass-extract-js options plugin name should NOT convert keys to camelCase 1`] = `
{
  "MY-VAR": "500px",
  "My-Cool_var": "400px",
  "my-var": "200px",
  "myVar": 0,
  "my__var--3": "700px",
  "my_other_vAr": "300px",
  "my_var": "200px",
  "my_var_2": "600px",
  "value-Map": {
    "my-var5": "900px",
    "my__var--6": "1000px",
    "my_var4": "800px",
  },
}
`;

exports[`sass-extract-js should convert basic SASS vars 1`] = `
{
  "boolean": true,
  "builtinFunc": "rgb(255, 51, 51)",
  "calculated": 200,
  "customFunc": "rgb(255, 0, 0)",
  "hexColor": "rgb(240, 240, 240)",
  "importantStyle": "10px !important",
  "inline": 100,
  "mixedList": "1px solid rgb(0, 0, 0)",
  "mixinA": "mixin-variable",
  "mixinB": 456,
  "namedColor": "rgb(255, 0, 0)",
  "nestedGlobal": "100px",
  "null": null,
  "number": 100,
  "numberList": "0 10px 1em 10%",
  "numberWithUnit": "100px",
  "quotedString": "quoted",
  "rgbColor": "rgb(255, 255, 255)",
  "rgbaColor": "rgba(255, 255, 255, 0.5)",
  "stringList": "'Helvetica', 'Arial', sans-serif",
  "unquotedString": "string",
  "valueMap": {
    "keyOne": "2em",
    "keyThree": false,
    "keyTwo": "key",
  },
}
`;

exports[`sass-extract-js should convert keys to camelCase 1`] = `
{
  "myCoolVar": "400px",
  "myOtherVAr": "300px",
  "myVar": "500px",
  "myVar2": "600px",
  "myVar3": "700px",
  "valueMap": {
    "myVar4": "800px",
    "myVar5": "900px",
    "myVar6": "1000px",
  },
}
`;

exports[`sass-extract-js should handle file with no variables 1`] = `{}`;

exports[`sass-extract-js should handle imports 1`] = `
{
  "importedVar": true,
  "myVar": "100px",
}
`;

exports[`sass-extract-js should pass sass options through 1`] = `
{
  "myVar": "100px",
  "nestedVar": false,
}
`;
